packages/google_maps_flutter/lib/src/google_maps_controller.dart

```diff
@@ -143,6 +142,58 @@ class GoogleMapController {
     // _markersController = MarkersController(stream: this._streamController);
   }
 
+  JavascriptChannel _onBoundsChanged() {
+    return JavascriptChannel(
+        name: 'BoundChanged',
+        onMessageReceived: (JavascriptMessage message) async {
+          final LatLng center = await getCenter();
+          final double zoom = await getZoomLevel();
+          if (!_mapIsMoving) {
+            _mapIsMoving = true;
+            _streamController.add(CameraMoveStartedEvent(_mapId));
+          }
+          _streamController.add(
+            CameraMoveEvent(_mapId, CameraPosition(target: center, zoom: zoom)),
+          );
+        });
+  }
+
+  JavascriptChannel _onIdle() {
+    return JavascriptChannel(
+        name: 'Idle',
+        onMessageReceived: (JavascriptMessage message) async {
+          _mapIsMoving = false;
+          _streamController.add(CameraIdleEvent(_mapId));
+        });
+  }
+
+  JavascriptChannel _onClick() {
+    return JavascriptChannel(
+        name: 'Click',
+        onMessageReceived: (JavascriptMessage message) async {
+          final Map<String, dynamic> event =
+              json.decode(message.message) as Map<String, dynamic>;
+          assert(event['latLng'] != null);
+          final LatLng position = LatLng(event['latLng']['lat'] as double,
+              event['latLng']['lng'] as double);
+          _streamController.add(MapTapEvent(_mapId, position));
+        });
+  }
+
+  JavascriptChannel _onRightClick() {
+    // NOTE: LWE does not support a long press event.
+    return JavascriptChannel(
+        name: 'RightClick',
+        onMessageReceived: (JavascriptMessage message) async {
+          final Map<String, dynamic> event =
+              json.decode(message.message) as Map<String, dynamic>;
```

Comment:
Can you always be sure of the format of the message?
if you didn't, how about this. I haven't checked whether the below works, but please refer to the meaning only.
```suggestion
              dynamic temp =  json.decode(message.message);
              if (temp is Map<String, dynamic> ) { 
                  event =  temp as Map<String, dynamic>;
              } 
```

Reply:
It's a so good idea. However, I think it will be always returned into JSON style if it works properly.